xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -1020,14 +1020,35 @@ bool CVideoPlayer::ReadPacket(DemuxPacket*& packet, CDemuxStream*& stream)
   }
 
   // read a data frame from stream.
-  if (m_pDemuxer)
+  if (m_postPonedPacket)
+  {
+    packet = m_postPonedPacket;
+    m_postPonedPacket = nullptr;
+  }
+  else if (m_pDemuxer)
     packet = m_pDemuxer->Read();
 
   if (packet)
   {
     // stream changed, update and open defaults
     if (packet->iStreamId == DMX_SPECIALID_STREAMCHANGE)
     {
+      if (m_CurrentVideo.id >= 0 && m_CurrentAudio.id >= 0 && m_VideoPlayerVideo &&
+          m_VideoPlayerAudio && m_VideoPlayerVideo->HasData() &&
+          m_VideoPlayerAudio->HasData() &&
+          ((m_CurrentVideo.syncState != IDVDStreamPlayer::SYNC_INSYNC ||
```

Comment:
reason for two `((` instead of one `(`?

Reply:
Also, the format of this complicated logic block is how clang-tidy would present it, but a more human-based format would increase understanding. Here is what I would do:

```c++

// Check for AV IDs
if (m_CurrentVideo.id >= 0 && m_CurrentAudio.id)
{
  // Check for AV players
  if (m_VideoPlayerVideo && m_VideoPlayerVideo->HasData() &&
      m_VideoPlayerAudio && m_VideoPlayerAudio->HasData())
  {
    // Check AV sync state
    if (m_CurrentVideo.syncState != IDVDStreamPlayer::SYNC_INSYNC ||
        m_CurrentAudio.syncState != IDVDStreamPlayer::SYNC_INSYNC ||
        m_CurrentVideo.avsync != CCurrentStream::AV_SYNC_NONE ||
        m_CurrentAudio.avsync != CCurrentStream::AV_SYNC_NONE)
     {
        CLog::Log(LOGDEBUG, "...");
        ...
        return true;
      }
    }
  }

  m_SelectionStreams.Clear(STREAM_NONE, STREAM_SOURCE_DEMUX);
  ...
```

The above is more understandable to me. Sure, it increases nesting, but I'd argue that's an acceptable engineering tradeoff in this situation.